cp/cp_config.c

```diff
@@ -67,7 +67,22 @@ config_change_cbk(char *config_file, uint32_t flags)
 	 * For now I am just going to switch to new config. Anyway its just selection of DPs 
 	 */
 	struct app_config *old_config = appl_config;
-	appl_config = new_cfg; 
+    /* Copy resource pointers from old config to new config */
+	struct dp_info *dpNew; 
+	struct dp_info *dpOld; 
+	LIST_FOREACH(dpNew, &new_cfg->dpList, dpentries) {
+		LIST_FOREACH(dpOld, &old_config->dpList, dpentries) {
+			if(dpOld->dpId == dpNew->dpId) {
+                dpNew->s1u_sgw_ip = dpOld->s1u_sgw_ip;
+				dpNew->upf = dpOld->upf;
+				dpNew->static_pool_tree = dpOld->static_pool_tree;
+				dpNew->static_pool = dpOld->static_pool;
```

Comment:
Can we free up dynamically allocated resources of `dpNew` before this re-assignment (e.g. `dpNew->static_pool`)? Isn't `init_spgwc_dynamic_config()` initializing it?

Reply:
We are taking care of not to allow static address pool change on the fly. But logic was buggy and it was causing static address pool getting initialized again n again. I am fixing it.